<launch>

  <!-- LAUNCH RVIZ -->

  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <arg name="config" default="true" />
  <arg unless="$(arg config)" name="command_args" default="" />
  <arg     if="$(arg config)" name="command_args" default="-d $(find moveit_recorder)/launch/designit.rviz" />

  <include file="$(find pr2_moveit_config)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>

  <node pkg="tf" type="static_transform_publisher" name="virtual_joint_broadcaster_0" args="0 0 0 0 0 0 odom_combined base_footprint 100" />

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="/use_gui" value="false"/>
    <rosparam param="/source_list">[/move_group/fake_controller_joint_states]</rosparam>
  </node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen" />

  <include file="$(find pr2_moveit_config)/launch/move_group.launch">
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="fake_execution" value="true"/>
    <arg name="info" value="true"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="publish_monitored_planning_scene" value="false" />
  </include>

  <node name="planning_scene_throttler" type="throttle" pkg="topic_tools" args="messages /planning_scene 2" />  

  <node name="$(anon rviz)" launch-prefix="$(arg launch_prefix)" pkg="rviz" type="rviz" respawn="false"
      args="$(arg command_args)" output="screen">
    <rosparam command="load" file="$(find pr2_moveit_config)/config/kinematics.yaml"/>
  </node>
  
  <arg name="from_marker_topic" default="from_marker_state" />
  <arg name="from_marker_pose_topic" default="from_marker_pose" />
  <arg name="to_marker_topic" default="to_marker_state" />

  <arg name="display_robot_topic" default="interactive_robot_state" />
  <arg name="robot_marker_topic" default="interactive_robot_markers" />
  <arg name="interactive_marker_topic" default="interactive_robot_imarkers" />

  <node name="start_marker" pkg="moveit_recorder" type="markers_node" respawn="false" output="screen"
        args="--display_robot_topic $(arg display_robot_topic) 
              --robot_marker_topic $(arg robot_marker_topic) 
              --interactive_marker_topic $(arg interactive_marker_topic) 
              --from_marker_topic $(arg from_marker_topic) 
              --from_marker_pose_topic $(arg from_marker_pose_topic) 
              --to_marker_topic $(arg to_marker_topic)">
  </node>

  <arg name="host" default="127.0.0.1" />
  <arg name="port" default="33829" />
  <arg name="planning_scene_topic" default="planning_scene" />
  <arg name="save_dir" default="/tmp/" />
  <node name="designer" pkg="moveit_recorder" type="benchmark_designer_node" respawn="false" output="screen"
        args="--host $(arg host) 
              --port $(arg port) 
              --from_marker_topic $(arg from_marker_topic) 
              --from_marker_pose_topic $(arg from_marker_pose_topic) 
              --to_marker_topic $(arg to_marker_topic) 
              --save_dir $(arg save_dir) 
              --planning_scene_topic $(arg planning_scene_topic)">
  </node>
</launch>
